### DISCUSSION TITLE
How to make run-or-raise focus the last used window?

### DISCUSSION THREAD

**Discussion by BooAA** [stumpwm/stumpwm#1133] [Q&A]
Hi, thanks for the work on this amazing window manager. I would like to replace all of my desktop setup with stumpwm alternative. One of them is [jumpapp](https://github.com/mkropat/jumpapp), which will launch the application if it's not already running, or switch to them and  cycle through the windows. The closet thing I can find in stumpwm is  `defprogram-shortcut`, like below: 

```lisp
(defprogram-shortcut kitty
  :command "kitty"
  :props '(:class "kitty")
  :map *top-map*
  :key (kbd "s-t"))
```

But `defprogram-shortcut` seems to always focus the window in same order. Say I have kitty terminal A, B, C and the last focused one is B. Then I switch to other application and jump back, then A will always be focused instead of B. I would like to focus the last window before I jump to other application, how should I configure stumpwm to do that? Thanks.

**Comment by PuercoPop** [comment]
What `defprogram-shortcut` does is call `find-matching-windows`, which after filtering windows based on the props, sorts the windows by number. If I understand you correctly you want to apply the same filtering strategy but order the windows by focus order. That information isn't tracked. So you would have to keep track of the recently focused windows using the `*focus-window-hook*`.
